### PR TITLE
PS-8803 Update Upgrade topics for 8.1 (8.1)

### PR DESCRIPTION
**Pull request by patrickbirch**
new file:   docs/upgrade-percona-repos.md
	new file:   docs/upgrade-standalone-packages.md
	new file:   docs/upgrade-strategies.md
	new file:   docs/upgrade.md
	modified:   variables.yml

### CODE REVIEW

File: docs/clone-plugin.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# MySQL Clone plugin
+
+Introduced in MySQL 8.0.17, the MySQL Clone plugin lets you locally clone data or from a remote server as a physical snapshot of the data stored in InnoDB. This snapshot includes schemas, tables, tablespaces, and data dictionary metadata. The cloned data is a functional data directory and can be used for provisioning a server .
```
Comment (alina-derkach-oaza) [inline docs/clone-plugin.md]:
```suggestion
The MySQL Clone plugin lets you clone data locally or from a remote server as a physical snapshot of the data stored in InnoDB. This snapshot includes schemas, tables, tablespaces, and data dictionary metadata. The cloned data is a functional data directory and can be used for provisioning a server.
```

File: docs/clone-plugin.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+```
+
+The result lists the Clone plugin and the status.
+
+## Clone data
+
+The SQL statement used to clone data depends on if the operation is local or remote. The following code is an example of cloning data from a remote server:
+
+```{.bash data-prompt="mysql>"}
+mysql> CLONE INSTANCE FROM `root@remote.server:13336` IDENTIFIED BY `root`;
```
Comment (percona-ysorokin) [inline docs/clone-plugin.md]: May be change `root` to `user` for this example just to discourage users from using admin accounts for this operations.

File: docs/downgrade.md [hunk 3]
```diff
@@ -1,5 +1,23 @@
...
-Downgrading Percona Server for MySQL {{vers}} to 8.0 is not supported.
+Downgrading to a 5.7 or earlier series is not supported.
 
-Each Innovation release of Percona Server for MySQL can contain significant changes.
\ No newline at end of file
+The following table lists the downgrade paths:
+
+| Path | Examples | Methods |
+|---|---|---|
+| Within a bugfix release series or an LTS release | 8.0.37 to 8.0.34 | in-place downgrade, logical downgrade, asynchronous replication, MySQL Clone plugin |
```
Comment (percona-ysorokin) [inline docs/downgrade.md]: Why this particular version `8.0.37`?

File: docs/upgrade-components.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# Upgrading from plugins to components 
+
+The following plugins have changed:
+
+| Plugin | 8.0 information | {{vers}} changes | Notes |
+|---|---|---|---|
+| `keyring_vault` | Only available as a plugin | `component_keyring_vault` | A manual upgrade path is required. For example, the plugin configuration file, specified by the `keyring_vault_config` system variable, must be transformed to a JSON format for the `component_keyring_vault.cnf`. |
+| `audit_log` | Only available as a plugin | removed | Recommended that you use the `component_audit_log_filter`. |
+| `audit_log_filter` | Only available as a plugin | `component_audit_log_filter` | the manual upgrade path required. |
+| `data_masking` | Available as a plugin and component | `component_masking_functions` | Migrate to the plugin to the component in 8.0 before the upgrade to {{vers}} |
```
Comment (percona-ysorokin) [inline docs/upgrade-components.md]:
```suggestion
| `data_masking` | Available as a plugin and component | `component_masking_functions` | Migrate the plugin to the component in 8.0 before the upgrade to {{vers}} |
```

File: docs/upgrade-percona-repos.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+    3. Modify the database configuration file, `my.cnf`, as needed.
+
+    4. Install Percona Server for MySQL:
+
+        ```{.bash data-prompt="$"}
+        $ sudo apt update
+        $ sudo apt install curl
+        $ curl -0 https://repo.percona.com/apt/percona-release_latest.generic_all.deb 
+        $ sudo apt install gnupg2 lsb-release ./percona-release_latest.generic_all.deb
```
Comment (percona-ysorokin) [inline docs/upgrade-percona-repos.md:34]: Not sure if `percona-release_latest` will include innovative releases (ie 8.1.0)

File: docs/upgrade-percona-repos.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        $ sudo apt update
+        $ sudo percona-release setup ps80
```
Comment (percona-ysorokin) [inline docs/upgrade-percona-repos.md]: `ps80` will not include innovative releases.
The last time we discussed this the name should be `ps-8x-innovative` but this needs to be clarified with Evgeniy's team.

File: docs/upgrade-percona-repos.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+         ```{.bash data-prompt="$"}
+         $ rpm -qa | grep '^mysql-' | xargs rpm -e --nodeps
+         ```
+
+
+    6. Install the `percona-server-server` package:
+
+         ```{.bash data-prompt="$"}
+         $ sudo yum install https://repo.percona.com/yum/percona-release-latest.noarch.rpm
+         $ sudo percona-release setup ps{{pkg}}
```
Comment (percona-ysorokin) [inline docs/upgrade-percona-repos.md]: Again, probably something like `ps-8x-innovation`

File: docs/upgrade-standalone-packages.md [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+
+        * `percona-server-server`
+
+        * `percona-server-client`
+
+        * `percona-server-common`
+
+        * `libperconaserverclient21`
+
+        The following example downloads Percona Server for MySQL 8.0.29-21 release
```
Comment (percona-ysorokin) [inline docs/upgrade-standalone-packages.md]: Here  and below, we need to show an example of upgrading to `8.1.0-1`, right?

File: docs/upgrade-standalone-packages.md [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+        This command installs only packages required to run the Percona Server for MySQL
+        {{vers}}.
+
+    6. You can install all the packages (for debugging, testing, etc.) with `sudo rpm -ivh \*.rpm`.
+
+        !!! note
+        
+            When manually installing packages, you must resolve all the dependencies and install missing ones.
+
+    7. Modify your configuration file, `my.cnf`, and install the plugins if necessary. If you are using the TokuDB storage engine you must comment out all the TokuDB specific variables in your configuration file(s) before starting the server, otherwise, the server will not start.
```
Comment (percona-ysorokin) [inline docs/upgrade-standalone-packages.md]: TokuDB references should be removed

**Review by percona-ysorokin — CHANGES_REQUESTED** [review]

File: docs/upgrade-percona-repos.md (hunk 5 above)
Comment (patrickbirch) [reply docs/upgrade-percona-repos.md:34]: Neither am I - can fix it after the release, if needed.

File: docs/upgrade-percona-repos.md (hunk 6 above)
Comment (patrickbirch) [reply docs/upgrade-percona-repos.md]: Right - all of this is unknown at this time. Can fix after the release, as needed.

File: docs/upgrade-percona-repos.md (hunk 7 above)
Comment (patrickbirch) [reply docs/upgrade-percona-repos.md]: Can fix after the release, as needed.

File: docs/upgrade-standalone-packages.md (hunk 8 above)
Comment (patrickbirch) [reply docs/upgrade-standalone-packages.md]: We are planning to update the docs with examples after the release.

File: docs/upgrade-percona-repos.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+
+    Run the following commands as root or use the `sudo` command.
+
+
+    1. Make a full backup (or dump if possible) of your database. Move the database configuration file, `my.cnf`, to another directory to save it. If the configuration file is not moved, it can be overwritten.
+
+
+    2. Stop the server with the appropriate command for your system:
+   
+       ```{.bash data-prompt="$"}
```
Comment (alina-derkach-oaza) [inline docs/upgrade-percona-repos.md:22]:
```suggestion
        ```{.bash data-prompt="$"}
```

File: docs/upgrade-percona-repos.md [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+       systemctl stop mysql`
```
Comment (alina-derkach-oaza) [inline docs/upgrade-percona-repos.md:23]:
```suggestion
        systemctl stop mysql`
```

File: docs/upgrade-percona-repos.md [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+       ```
```
Comment (alina-derkach-oaza) [inline docs/upgrade-percona-repos.md:24]:
```suggestion
        ```
```

File: docs/upgrade-standalone-packages.md [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+    skip-grant-tables
+    ```
+    
+     Restart the mysql server with `service mysql start`. 
+
+    9.  Running the upgrade:
+
+        The mysqld binary automatically runs the upgrade process. To find more information, see [MySQL Upgrade Process Upgrades](https://dev.mysql.com/doc/refman/{{vers}}/en/upgrading-what-is-upgraded.html)
+
+    10. Restart the server with `service mysql restart`. After the service has been successfully restarted you can use the new Percona Server for MySQL {{vers}}.
```
Comment (alina-derkach-oaza) [inline docs/upgrade-standalone-packages.md]: The list is broken starting from step 7.

File: docs/upgrade-components.md [hunk 14]
```diff
@@ -0,0 +1,42 @@
+# Upgrading from plugins to components 
+
+The following plugins have changed:
+
+| Plugin | 8.0 information | {{vers}} changes | Notes |
+|---|---|---|---|
+| `keyring_vault` | Only available as a plugin | `component_keyring_vault` | A manual upgrade path is required. For example, the plugin configuration file, specified by the `keyring_vault_config` system variable, must be transformed to a JSON format for the `component_keyring_vault.cnf`. |
+| `audit_log` | Only available as a plugin | removed | Recommended that you use the `component_audit_log_filter`. |
+| `audit_log_filter` | Only available as a plugin | `component_audit_log_filter` | Migrate the plugin to the component in 8.0 before the upgrade to {{vers}}. |
+| `data_masking` | Available as a plugin and component | `component_masking_functions` | Migrate to the plugin to the component in 8.0 before the upgrade to {{vers}} |
```
Comment (percona-ysorokin) [inline docs/upgrade-components.md]: Not addressed.
Still `Migrate to the plugin to the component`

**Review by percona-ysorokin — APPROVED** [review]
LGTM with one typo fixed

**Pull request merged**